app/dashboard/static/js/app/view-builds-job-branch-kernel.2017.7.1.js

```diff
@@ -326,6 +329,15 @@ require([
                 archLabelNode.appendChild(document.createTextNode(arch));
             }
 
+            if (compiler !== null) {
```

Comment:
There should always be a `build_environment` field so no need to check if it's `null`.

Reply:
is it makes sense for other variables? For example, as used in `arch`.

Reply:
I see there is a test for `arch`, I guess it's fine to leave it like that in this PR.  But yes, in principle the `arch` field should always be defined in build documents.

We could have a clean-up PR later once the main changes have been made to review and improve the code with things like this.

Reply:
I think that it will be a good to do! Split functions, make some code that already is replicable in a reusable, improve patterns in variable names, etc :)